network/server/tchannelthrift/node/service.go

```diff
@@ -478,6 +485,139 @@ func (s *service) FetchBlocksMetadataRaw(tctx thrift.Context, req *rpc.FetchBloc
 	return result, nil
 }
 
+func (s *service) FetchBlocksMetadataRawV2(tctx thrift.Context, req *rpc.FetchBlocksMetadataRawV2Request) (*rpc.FetchBlocksMetadataRawV2Result_, error) {
+	if s.db.IsOverloaded() {
+		s.metrics.overloadRejected.Inc(1)
+		return nil, tterrors.NewInternalError(errServerIsOverloaded)
+	}
+
+	var err error
+	callStart := s.nowFn()
+	defer func() {
+		if err != nil {
+			s.metrics.fetchBlocksMetadata.ReportSuccess(s.nowFn().Sub(callStart))
+		}
+	}()
+
+	ctx := tchannelthrift.Context(tctx)
+	if req.Limit <= 0 {
+		return nil, nil
+	}
+
+	var opts block.FetchBlocksMetadataOptions
+	if req.IncludeSizes != nil {
+		opts.IncludeSizes = *req.IncludeSizes
+	}
+	if req.IncludeChecksums != nil {
+		opts.IncludeChecksums = *req.IncludeChecksums
+	}
+	if req.IncludeLastRead != nil {
+		opts.IncludeLastRead = *req.IncludeLastRead
+	}
+
+	var (
+		nsID      = s.newID(ctx, req.NameSpace)
+		start     = time.Unix(0, req.RangeStart)
+		end       = time.Unix(0, req.RangeEnd)
+		pageToken = &pt.PageToken{}
+	)
+	err = proto.Unmarshal(req.PageToken, pageToken)
+	if err != nil {
+		return nil, tterrors.NewBadRequestError(errInvalidPageToken)
+	}
+
+	fetchedMetadata, nextShardIndex, err := s.db.FetchBlocksMetadata(
```

Comment:
Would this `err` get a new definition at this block level?  i.e. not 100% sure that the defer would capture if this `err` is not nil or not, probably can test with a simple playground: